sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala

```diff
@@ -37,9 +37,14 @@ import org.apache.spark.sql.streaming.{GroupStateTimeout, OutputMode}
  */
 object UnsupportedOperationChecker extends Logging {
 
-  def checkForBatch(plan: LogicalPlan): Unit = {
+  def checkForBatch(plan: LogicalPlan, isSqlContext: Boolean): Unit = {
     plan.foreachUp {
       case p if p.isStreaming =>
+        if (isSqlContext) {
+          throwError("SQL streaming queries must be executed with a Spark Declarative Pipeline. " +
```

Comment:
How is SQL streaming queries different from Scala/Python streaming queries that we don't want to run it directly?